accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,332 @@
+In the current design, we are exposing metrics via Prometheus and making them available in Grafana, and more importantly we are exposing log messages via Loki/LogCLI to CLI and Grafana users. It is important to notice that after running this tool, and until related containers are destroyed, the Grafana Dashboards (and other components like Prometheus and Loki/LogCLI) are exposing metrics and log messages that may contain sensitive data and information to any non-root user on the system or to anyone that have access to this host in the network.
+
+The Promtail pipeline definition can be enhanced to use the [replace](https://grafana.com/docs/loki/latest/send-data/promtail/stages/replace/) stage in order to hide sensitive data from log files. Users must have a configuration file to define the different filters to apply to the Promtail pipeline. Filters must be also configurable via the Monitoring Formula.
+
+TLS must be enabled for Promtail, Prometheus and Loki to ensure a secure transport.
+
+An authentication backend for Prometheus and Loki is usually delegated to an authentication reverse proxy. This must be included in the documentation and probably be considered in the Formula.
+
+
+# Learning Curve and Documentation
+
+System administrators and developers may need to familiarize themselves with Loki/Promtail's log querying language (LogQL) and Grafana's alerting mechanisms. The learning curve associated with these tools could slow down initial adoption and efficiency gains.
+
+In this sense, we must provide a good documentation with tutorials, use cases and examples to help the users to early and smoothly adopt the new monitoring capabilities.
+
+
+# Implementation steps
+
+### Phase 0: Disconnected Setup
+
+#### Uyuni Health Check
+
+- Create RPM packages in OBS/IBS.
+- Use container images building in OBS/IBS for the different components.
+- Use alerting rules to detect potential issues.
+- Gathers opinions from engineers about possible improvements of metrics, dashboards and alerts.
+
+#### Saline
+
+- Research about using TCP instead of IPC sockets for Salt Master internal sockets.
+
+### Phase 1: Integrated Solution
+
+#### Uyuni Health Check
+
+- Integration with the Monitoring Stack. Enhance the monitoring formula to enable Health Check monitoring.
+- Enhance metrics, dashboards and alerts.
+- Allow users to enhance the Promtail pipeline to add filters and/or replace patterns to be able to replace sensitive information.
+- Use the alert system to detect potential issues and provide recommendations.
+
+#### Saline
+
+- Build Saline in OBS/IBS and make it part of the Uyuni Server image
+- Saline runs inside the `uyuni-server` container.
+
+### Phase 2: Standalone Solution
+
+#### Uyuni Health Check
+
+- Provide a "standalone" version only (no integration with the Monitoring stack) via an Uyuni server channel (or others so it is available for engineers or supporters).
+- Reuse configuration from the "integrated" version.
```

Comment:
In the discussion we have brought up an idea of having a _one button_ default configuration deployment of standard monitoring/logging stack. I think we could integrate it in the SUMA Admin UI.

Do you think we could add a comment about this alternative implementation?